dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -37,148 +43,160 @@
+
+		Bundle bundle = null;
+
+		try (InputStream bundleStream = request.getInputStream()) {
+
+			final String bundlePath         = ConfigUtils.getBundlePath()+ File.separator + MY_TEMP;
+			final String bundleFolder       = fileName.substring(0, fileName.indexOf(".tar.gz"));
+			final PublishAuditStatus status = PublishAuditAPI.getInstance().updateAuditTable(
+					remoteIP, remoteIP, bundleFolder, true);
+
+			// save bundle if it doesn't exists
+			bundle = APILocator.getBundleAPI().getBundleById(bundleFolder);
+			if (bundle == null || bundle.getId() == null) {
+
+				bundle = new Bundle();
+				bundle.setId(bundleFolder);
+				bundle.setName(fileName.replace(".tar.gz", ""));
+				bundle.setPublishDate(Calendar.getInstance().getTime());
+				bundle.setOwner(APILocator.getUserAPI().getSystemUser().getUserId());
+				bundle.setForcePush(forcePush);
+				APILocator.getBundleAPI().saveBundle(bundle);
+			}
+
+			//Write file on FS
+			FileUtil.writeToFile(bundleStream, bundlePath + fileName);
 
-        //My key
-        Optional<String> myKey=PushPublisher.retriveEndpointKeyDigest(mySelf);
-        if(!myKey.isPresent()) {
-          return false;
-        }
+			//Start thread
 
+			if(!status.getStatus().equals(Status.PUBLISHING_BUNDLE)) {
 
-        return token.equals( myKey.get() );
-    }
+				DotConcurrentFactory.getInstance()
+						.getSubmitter()
+						.submit(new PublishThread(fileName, null, null, status));
+			}
 
+			return bundle;
+		} catch (Exception e) {
 
+			Logger.error(
+					PublisherQueueJob.class,
+					String.format("Error caused by remote call of: Remote IP - %s, bundle file name - %s, end point- %s",
+							remoteIP, fileName,  remoteIP));
+			Logger.error(PublisherQueueJob.class,e.getMessage(),e);
+			throw e;
+		}
+	}
+
+	private String getFileNameFromRequest(HttpServletRequest request) {
+		try {
+			final String fileNameValue = request.getHeader("Content-Disposition")
+					.split(";")[1]
+					.trim()
+					.split("=")[1];
+			return fileNameValue.substring(1, fileNameValue.length() - 1);
+		} catch (Exception e) {
+			return null;
+		}
+	}
+
+	private String generatedBundleFileName() {
+		return String.format("bundle_%d.tar.gz", System.currentTimeMillis());
+	}
+
+	private boolean isAdmin(final User user) {
+
+		return null != user && user.isBackendUser() && user.isAdmin();
+	}
+
+	/**
+	 * Validates a received token
+	 *
+	 * @param token    Token to validate
+	 * @param publishingEndPoint   Current end point
+	 * @return True if valid
+	 * @throws IOException If fails reading the security token
+	 */
+	public static boolean isValidToken (final String token,
+										final PublishingEndPoint publishingEndPoint) throws IOException {
+
+		//My key
+		final  Optional<String> endpointKeyDigest = PushPublisher.retriveEndpointKeyDigest(publishingEndPoint);
+		return endpointKeyDigest.isPresent()? token.equals( endpointKeyDigest.get() ): false;
```

Comment:
Codacy found an issue: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)